packages/extension/src/components/pages/CreatePassword/CreatePassword.tsx

```diff
@@ -24,18 +24,19 @@ export const CreatePassword: FC<CreatePasswordProps> = ({
 }) => {
   const [passwordError, setPasswordError] = useState('');
   const [saveWalletError, setSaveWalletError] = useState('');
+  const passwordRef = useRef<HTMLInputElement | null>(null);
+  const confirmPasswordRef = useRef<HTMLInputElement | null>(null);
 
   const handleNext = useCallback(() => {
-    const passwordValue = (document.getElementById('password') as HTMLInputElement).value;
-    const confirmPasswordValue = (document.getElementById('confirm-password') as HTMLInputElement)
-      .value;
-    if (passwordValue.length < 8) {
+    let passwordValue: string | undefined = passwordRef.current?.value;
+    let confirmPasswordValue: string | undefined = confirmPasswordRef.current?.value;
+    if (passwordValue === undefined || passwordValue == null || passwordValue.length < 8) {
       setPasswordError('Password must be at least 8 characters long');
     } else if (passwordValue !== confirmPasswordValue) {
       setPasswordError('Passwords must match');
     } else {
       try {
-        saveWallet(wallet, passwordValue);
+        saveWallet(wallet, confirmPasswordValue);
```

Comment:
In this algorithm, you do make most of the check with `passwordValue`, I would keep saving the waller with: `passwordValue`.
This is really minor, it's not gonna change the behavior or logic anyways.
```suggestion
        saveWallet(wallet, passwordValue);
```